### PR TITLE
Daps Training - clustering vectors

### PR DESCRIPTION
**Pull request by kstathou**
Draft PR for this week's tasks.

- [x] Clone `dev`, create a new branch with a short sensible name, prefixed with this issue number
- [x] Create an empty package under `nesta.packages.nlp_utils` called `text2vec`
- [x] Commit and create a *draft* PR for this branch
- [x] Use TTD to write functional code to transform text data into vectors. Where possible, support batching/chunking of data if this leads to performance gains. The output vectors should be written out in chunks on S3.

### CODE REVIEW

File: nesta/packages/nlp_utils/embed_clustering.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import numpy as np
+from sklearn.mixture import GaussianMixture
+
+np.random.seed(42)
+
+
+def vals2arr(d):
```
Comment (jaklinger) [inline nesta/packages/nlp_utils/embed_clustering.py]: where is this used?

File: nesta/packages/nlp_utils/embed_clustering.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    # For now, GMMs hyperparameters are hardcoded.
+    gmm = GaussianMixture(covariance_type='tied', init_params='kmeans', max_iter=100,
+                          means_init=None, n_components=80, n_init=1, precisions_init=None,
+                          random_state=42, reg_covar=1e-06, tol=0.001, verbose=0,
+                          verbose_interval=10, warm_start=False, weights_init=None)
+    gmm.fit(vectors)
+    return gmm
+
+
+def filter_arr(arr, thresh=.2):
```
Comment (jaklinger) [inline nesta/packages/nlp_utils/embed_clustering.py]: this might be better in a general purpose `np_utils.py` in a function titled `arr2dict` with an optional arg `thresh` like so:

```
def arr2dic(arr, thresh=None):
    if thresh is not None:
        ids = np.where(arr <= thresh)[0]
    else:
        # get all the ids
     # return the ids
```

File: nesta/packages/nlp_utils/tests/test_embedding_clustering.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+def test_vals2arr():
+    arr = np.random.rand(3, 2)
+    d = {i: arr[i] for i in range(arr.shape[0])}
+    _, arrays = vals2arr(d)
+    assert arrays.shape == arr.shape
+
+
+def test_clustering():
+    arr = np.random.rand(1000, 3)
```
Comment (jaklinger) [inline nesta/packages/nlp_utils/tests/test_embedding_clustering.py]: i would add other tests for approx reproducibility with different random seeds

File: nesta/packages/nlp_utils/tests/test_embedding_clustering.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+
+def test_vals2arr():
+    arr = np.random.rand(3, 2)
+    d = {i: arr[i] for i in range(arr.shape[0])}
+    _, arrays = vals2arr(d)
+    assert arrays.shape == arr.shape
+
+
+def test_clustering():
```
Comment (jaklinger) [inline nesta/packages/nlp_utils/tests/test_embedding_clustering.py]: the test name should tell you exactly what is being tested

File: nesta/packages/nlp_utils/tests/test_text2vec.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+def test_process_to_IDs_in_sparse_format():
+    documents = ['Water under the bridge.', 'Who let the dogs out.']
+    module = hub.Module("https://tfhub.dev/google/universal-sentence-encoder-lite/2")
+    with tf.compat.v1.Session() as sess:
+        spm_path = sess.run(module(signature="spm_path"))
+        sp = spm.SentencePieceProcessor()
+        sp.Load(spm_path)
+        values, indices, dense_shape = process_to_IDs_in_sparse_format(sp, documents)
+    assert(dense_shape[0] == len(documents))
```
Comment (jaklinger) [inline nesta/packages/nlp_utils/tests/test_text2vec.py]: add some other assertions on the array contents

File: nesta/packages/nlp_utils/text2vec.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    Universal Sentence Encoder (lite) model.
+
+    Args:
+        documents (:obj:`list` of :obj:`str`): List of raw text documents.
+
+    Returns:
+        doc_embeddings (:obj:`numpy.array` of :obj:`numpy.array` of :obj:`float`): Vector representation of documents.
+
+    """
+    module = hub.Module("https://tfhub.dev/google/universal-sentence-encoder-lite/2")
```
Comment (jaklinger) [inline nesta/packages/nlp_utils/text2vec.py]: add the URL as a global variable

Comment (jaklinger) [inline nesta/packages/nlp_utils/text2vec.py]: add the URL as a global variable

File: nesta/packages/nlp_utils/text2vec.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        sess.run([tf.compat.v1.global_variables_initializer(), tf.compat.v1.tables_initializer()])
+        doc_embeddings = sess.run(
+                                encodings,
+                                feed_dict={input_placeholder.values: values,
+                                           input_placeholder.indices: indices,
+                                           input_placeholder.dense_shape: dense_shape})
+    return doc_embeddings
+
+
+if __name__ == '__main__':
```
Comment (jaklinger) [inline nesta/packages/nlp_utils/text2vec.py]: remove this now your dev is finished

**Comment by jaklinger** [comment]
@kstathou just a nudge so that this doesn't go stale

File: nesta/packages/nlp_utils/embed_clustering.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
+import numpy as np
+from sklearn.mixture import GaussianMixture
+
+np.random.seed(42)
```
[the remark this answers is not in the document]
Comment (kstathou) [reply nesta/packages/nlp_utils/embed_clustering.py]: Is this for the `np.random.seed(42)`? Why should it be up to the user?

**Comment by jaklinger** [comment]
Because it will conflict with their own random seed if they use / implement
another algorithm

File: nesta/packages/nlp_utils/embed_clustering.py (hunk 2 above)
Comment (kstathou) [reply nesta/packages/nlp_utils/embed_clustering.py]: Created a new `np_utils.py` file under the `misc_utils`
